common/workunit/workunit.cpp

```diff
@@ -2487,8 +2487,74 @@ class CompoundAggregator : implements StatisticAggregator
     IArrayOf<StatisticAggregator> aggregators;
 };
 
+// Conditionally aggregate costs by
+// * excluding hThor costs when excludehThor=true
+// * or hThor costs that are children of workflow that already has a cost
+class FilteredCostAggregator : public CInterfaceOf<IWuScopeVisitor>
+{
+public:
+    FilteredCostAggregator(bool _excludeHThor) : excludeHThor(_excludeHThor) {}
 
-//To calculate aggregates, create a scope iterator, and an instance of a StatisticAggregator, and play the attributes through the interface
+    virtual void noteStatistic(StatisticKind kind, unsigned __int64 value, IConstWUStatistic & extra) override
+    {
+        if (extra.getCreatorType() == SCThthor)
+        {
+            if (excludeHThor)
+                return;
+            if (haveWorkFlowCost)
```

Comment:
I'm not sure why this is hthor only.  It should also have been short-circuited with nextSibling() on the wofkflow item.